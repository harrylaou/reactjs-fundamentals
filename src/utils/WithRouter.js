import React from 'react'

const WithRouter = (MyComponent) => {
  const withRouter = (props, context) => (
    <MyComponent {...props} router={context.router} />
  )

  withRouter.contextTypes = {
    router: React.PropTypes.object.isRequired
  }

  return withRouter
}

export default WithRouter
