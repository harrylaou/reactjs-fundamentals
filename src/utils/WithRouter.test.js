import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import WithRouter from './WithRouter'

describe('<WithRouter />', () => {
  const DemoElem = () => (
    <div>Hello World</div>
  )

  const Elem = WithRouter(DemoElem)

  it('should render with default props', () => {

    const wrapper = shallow(
      <Elem />,
      { context: { router: { text: 'thatIstheRouter' } } }
    )
    expect(shallowToJson(wrapper.dive())).toMatchSnapshot()
  })

  it('should get router with default props', () => {

    const wrapper = shallow(
      <Elem />,
      { context: { router: { text: 'thatIstheRouter' } } }
    )
    expect(wrapper.prop('router')).toEqual({ text: 'thatIstheRouter' })
  })
})