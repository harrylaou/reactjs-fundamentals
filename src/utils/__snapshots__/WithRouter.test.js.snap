// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithRouter /> should render with default props 1`] = `
<div>
  Hello World
</div>
`;
